src/WorldStorage/WSSAnvil.cpp

```diff
@@ -1472,17 +1472,48 @@ cBlockEntity * cWSSAnvil::LoadSignFromNBT(const cParsedNBT & a_NBT, int a_TagIdx
 
 
 
+bool cWSSAnvil::CheckEntityType(const char * a_IDTag, size_t a_IDTagLength, const AStringVector & a_ExpectedTypes)
+{
+	for (auto & et : a_ExpectedTypes)
+	{
+		if (strncmp(a_IDTag, et.c_str(), a_IDTagLength) == 0)
+		{
+			return true;
+		}
+	}
+	return false;
+}
+
+
+
+
+
 void cWSSAnvil::LoadEntityFromNBT(cEntityList & a_Entities, const cParsedNBT & a_NBT, int a_EntityTagIdx, const char * a_IDTag, size_t a_IDTagLength)
 {
-	if (strncmp(a_IDTag, "Boat", a_IDTagLength) == 0)
+	static const EntityLoaderMap EntityTypeToFunction
+	{
+		{ "Boat", &cWSSAnvil::LoadBoatFromNBT },
+		{ "minecraft:boat", &cWSSAnvil::LoadBoatFromNBT }
+
+	};
+
+	auto it = EntityTypeToFunction.find(a_IDTag);
```

Comment:
`a_IDTag` is not NUL-terminated, so you'll need an explicit size when converting to `AString`. Use `...find(AString(a_IDTag, a_IDTagLength));`